__tests__/components/HTMLBlock.test.tsx

```diff
@@ -0,0 +1,64 @@
+import { render, screen, cleanup } from '@testing-library/react';
+import React from 'react';
+import { renderToStaticMarkup, renderToString } from 'react-dom/server';
+import { vi } from 'vitest';
+
+import HTMLBlock from '../../components/HTMLBlock';
+import { compile, run } from '../../index';
+
+describe('HTML Block', () => {
+  beforeEach(() => {
+    global.mockFn = vi.fn();
+  });
+
+  afterEach(() => {
+    cleanup();
+    vi.restoreAllMocks();
+  });
+
+  it('runs user scripts in compat mode', () => {
+    render(<HTMLBlock runScripts={true}>{`<script>mockFn()</script>`}</HTMLBlock>);
+    expect(global.mockFn).toHaveBeenCalledTimes(1);
+  });
+
+  it("doesn't run user scripts by default", () => {
+    render(<HTMLBlock>{`<script>mockFn()</script>`}</HTMLBlock>);
+    expect(global.mockFn).toHaveBeenCalledTimes(0);
+  });
+
+  it("doesn't render user scripts by default", () => {
+    render(<HTMLBlock>{`<script>mockFn()</script>`}</HTMLBlock>);
+    expect(screen.queryByText('mockFn()')).not.toBeInTheDocument();
+  });
+
+  it("doesn't render user scripts with weird endings", () => {
+    render(<HTMLBlock>{`<script>mockFn()</script foo='bar'>`}</HTMLBlock>);
+    expect(screen.queryByText('mockFn()')).not.toBeInTheDocument();
+  });
+
+  it("doesn't render user scripts with a malicious string", () => {
+    render(<HTMLBlock>{`<scrip<script></script>t>mockFn()</s<script></script>cript>`}</HTMLBlock>);
+    expect(screen.queryByText('mockFn()')).not.toBeInTheDocument();
+  });
+
+  it("doesn't run scripts on the server (even in compat mode)", () => {
+    const html = `
+    <h1>Hello World</h1>
+    <script>mockFn()</script>
+    `;
+    const elem = <HTMLBlock runScripts={true}>{html}</HTMLBlock>;
+    const view = renderToString(elem);
+    expect(elem.props.runScripts).toBe(true);
+    expect(view.indexOf('<script>')).toBeLessThan(0);
+    expect(view.indexOf('<h1>')).toBeGreaterThanOrEqual(0);
+  });
+
+  it('renders the html in a `<pre>` tag if safeMode={true}', async () => {
+    const md = '<HTMLBlock safeMode={true}>{`<button onload="alert(\'gotcha!\')"/>`}</HTMLBlock>';
+    const code = compile(md);
+    const Component = await run(code);
+    expect(renderToStaticMarkup(<Component />)).toMatchInlineSnapshot(
+      '"<pre class="html-unsafe"><code>&lt;button onload=&quot;alert(&#x27;gotcha!&#x27;)&quot;/&gt;</code></pre>"',
+    );
+  });
```

Comment:
I _think_ we can scrap all these safety features now that all of the doc is executable?

Reply:
lol honestly all these tests helped me dial the component in. i didn't realize i had swapped the "cleaned" vs "dirty" html until i got to the veeeery last test

Reply:
Keep whatever tests, but I think we can wholesale remove `safeMode` and `runScripts`

Reply:
i was actually going to ask you about that -- `runScripts` is a component prop, but `safeMode` and `lazyImages` are both processor options/settings for the user. we still need to support these in a similar way, yeah? i tried making the components basically HOC and passing args that way, but `useMDXComponents` was Not Having It

Reply:
Instead of HOC's we should probably create a context for the options?